docs/federate.md

```diff
@@ -0,0 +1,130 @@
+There are two main reasons to deploy local query federation:
```

Comment:
Lines 16ff just below mention that. We also talk about ways to not federate on the infrastructure page. Does that address your point or do you think we need an explicit section here when you should **not** use federation?

Reply:
Hm, that's true. I guess I was thinking from the perspective of someone who is only interested in federated queries and jumps straight to this page.

One option could be to move lines 16-19 to the top of this page, so that a user can see upfront that, without having to set up anything locally, they can already run federated queries over Neurobagel's public datasets. Wdyt?

Reply:
Yeah, I think I would do that as a "contraindications" section